amf-graphql/shared/src/main/scala/amf/graphql/internal/spec/domain/GraphQLDirectiveDeclarationParser.scala

```diff
@@ -0,0 +1,56 @@
+package amf.graphql.internal.spec.domain
+
+import amf.core.client.scala.model.domain.extensions.{CustomDomainProperty, PropertyShape}
+import amf.graphql.internal.spec.context.GraphQLWebApiContext
+import amf.graphql.internal.spec.parser.syntax.TokenTypes._
+import amf.graphql.internal.spec.parser.syntax.{GraphQLASTParserHelper, Locations}
+import amf.shapes.client.scala.model.domain.NodeShape
+import org.mulesoft.antlrast.ast.{Node, Terminal}
+
+case class GraphQLDirectiveDeclarationParser(node: Node)(implicit val ctx: GraphQLWebApiContext)
+    extends GraphQLASTParserHelper {
+  var directive: CustomDomainProperty = CustomDomainProperty(toAnnotations(node))
+
+  def parse(parentId: String): CustomDomainProperty = {
+    directive.adopted(parentId)
+    parseName()
```

Comment:
You first need to parse the name with is required for ID adoption. You can check that all the directives in is-input-type-directive-arguments.jsonld end up with the same ID and some IDs have nullwrapper and null in them

Reply:
that's true, I've put the adopt after the `parseName()` method.